src/llama_stack/providers/remote/prompts/mlflow/mlflow.py

```diff
@@ -0,0 +1,565 @@
+            prompt_version = version
+
+        # Check if this is the default version
+        is_default = await self._is_default_version(mlflow_name, prompt_version)
+
+        return Prompt(
+            prompt_id=prompt_id,
+            prompt=template,
+            version=prompt_version,
+            variables=variables,
+            is_default=is_default,
+        )
+
+    async def update_prompt(
+        self,
+        prompt_id: str,
+        prompt: str,
+        version: int,
+        variables: list[str] | None = None,
+        set_as_default: bool = True,
+    ) -> Prompt:
+        """Update prompt (creates new version in MLflow).
+
+        Args:
+            prompt_id: Llama Stack prompt ID
+            prompt: Updated prompt template
+            version: Current version being updated
+            variables: Updated variables list (auto-extracted if not provided)
+            set_as_default: Set new version as default
+
+        Returns:
+            Updated Prompt resource with incremented version
+
+        Raises:
+            ValueError: If current version not found or validation fails
+        """
+        import mlflow
+
+        # Auto-extract variables if not provided
+        if variables is None:
+            variables = self._extract_variables(prompt)
+        else:
+            # Validate variables
+            template_vars = set(self._extract_variables(prompt))
+            declared_vars = set(variables)
+            undeclared = template_vars - declared_vars
+            if undeclared:
+                raise ValueError(f"Template contains undeclared variables: {sorted(undeclared)}")
+
+        mlflow_name = self.mapper.to_mlflow_name(prompt_id)
+
+        # Get all versions to determine the latest and next version number
+        versions_response = await self.list_prompt_versions(prompt_id)
+        if not versions_response.data:
+            raise ValueError(f"Prompt {prompt_id} not found")
+
+        max_version = max(p.version for p in versions_response.data)
+
+        # Verify the provided version is the latest
+        if version != max_version:
+            raise ValueError(
+                f"Version {version} is not the latest version. Use latest version {max_version} to update."
+            )
+
+        new_version = max_version + 1
+
+        # Prepare metadata tags
+        tags = self.mapper.get_metadata_tags(prompt_id, variables)
+
+        # Register new version in MLflow
+        try:
+            mlflow.genai.register_prompt(
+                name=mlflow_name,
+                template=prompt,
+                commit_message=f"Updated from version {version} via Llama Stack",
+                tags=tags,
+            )
+            logger.info(f"Updated prompt {prompt_id} to version {new_version}")
+        except Exception as e:
+            logger.error(f"Failed to update prompt in MLflow: {e}")
+            raise
+
+        # Set as default if requested
+        if set_as_default:
```

Comment:
the latest incremented version is the default, a user has to explicitly set the non-latest version to make it the default.

Reply:
From the definition in line 299, that is the default behavior:

```python
    async def update_prompt(
        self,
        prompt_id: str,
        prompt: str,
        version: int,
        variables: list[str] | None = None,
        set_as_default: bool = True,
    ) -> Prompt:
```

Is a different behavior preferred?